src/21-pubsub/index.md

```diff
@@ -0,0 +1,116 @@
+---
+# this is the final blog post's id (used in the directory)
+# this id is part of the url, and should only contain:
+#   letters, numbers, dashes.
+id: 25-pubsub
+
+breadcrumbs:
+  - {name: "Take a look at pubsub on IPFS", link: "./" }
+
+# the date here should be set to the final publication date,
+# on the day it is published.
+date: 2017-05-16
+
+# this is the Title
+title: Take a look at pubsub on IPFS
+
+# this is the name of the main author(s)
+author: Jeromy Johnson
+
+# technical details required for the software, don't change these.
+baseurl: ..
+template: tmpl/layouts/post.html
+collection: posts
+---
+
+To make IPFS fast for datacenters, local area networks, and large p2p
+applications, we need a fast publish-subscribe system. [Publish-Subscribe](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern),
+called 'pubsub' for short, is a pattern often used to handle events in
+large-scale networks. 'Publishers' send messages classified by topic or content and
+'subscribers' receive only the messages they are interested in, all without direct
+connections between publishers and subscribers. This approach offers much greater
+network scalability and flexibility.
+
+We have recently merged a simple, experimental pubsub implementation into go-ipfs.
+The implementation is very far from the performance and security levels we expect
+from our long-term target. But already, this feature allows for some very useful and interesting new applications.
+
+In this post, I will explain how to get started using `ipfs pubsub` and share some
+current and future examples of applications.
+
+## Getting started with ipfs pubsub
+First, you'll need to enable the pubsub code. Make sure you're running a recent version
```

Comment:
Make sure to note that there is also a js implementation. Documentation on that will come soon.